### PR TITLE
azurerm_role_definition: don't add `scope` to `assignable_scopes`

### LINKED ISSUE #8577: role_defintion is now including the scope as part of the assignable_scopes causing constant removals
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.24
+ provider.azuread v1.0.0
+ provider.azuredevops v0.0.1
+ provider.azurerm v2.28.0
+ provider.null v2.1.2
+ provider.random v2.3.0
+ provider.vault v2.14.0

### Affected Resource(s)

* `azurerm_role_definition`

### Terraform Configuration Files

```hcl
resource "azurerm_role_definition" "example" {
  count              = var.resource_count > 0 ? 1 : 0
  name               = "example-iam"
  role_definition_id = random_uuid.example.result
  scope              = data.azurerm_subscription.primary.id

  description = "This is a sample custom rule set"

  permissions {
    actions          = module.permissions.allow
    not_actions      = module.permissions.deny
    data_actions     = []
    not_data_actions = []
  }

  assignable_scopes = [var.resource_group_id] 
}
```

### Expected Behavior

Assignable scopes should only include the var.resource_group_id or the assignable_scopes that are defined.

### Actual Behavior

After viewing the state file both the var.resource_group_id and the scope data.azurerm_subscription.primary.id are shown in the assignable_scopes. This causes each subsequent run to try and remove the data.azurerm_subscription.primary.id or whatever is in the scope reference.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

I was digging through the provider code and I believe the issue is here. Prior to 2.27 there was never a check for assignedScope as part of the assignableScopes. Here's the code from 2.26:
```go
func expandRoleDefinitionAssignableScopes(d *schema.ResourceData) []string {
	scopes := make([]string, 0)

	assignableScopes := d.Get("assignable_scopes").([]interface{})
	for _, scope := range assignableScopes {
		scopes = append(scopes, scope.(string))
	}

	return scopes
}
```

Now it's making a check to see if the assignableScopes contains the assignedScope which appears to actually be adding the scope now:

```go
func expandRoleDefinitionAssignableScopes(d *schema.ResourceData) []string {
	scopes := make([]string, 0)

	// The first scope in the list must be the target scope as it it not returned in any API call
	assignedScope := d.Get("scope").(string)
        // This is now adding the assignedScope to the list scopes
	scopes = append(scopes, assignedScope) 
	assignableScopes := d.Get("assignable_scopes").([]interface{})
	for _, scope := range assignableScopes {
		// Ensure the assigned scope is not duplicated in the list if also specified in `assignable_scopes`
		if scope != assignedScope {
			scopes = append(scopes, scope.(string))
		}
	}

	return scopes
}
```

### PR DESCRIPTION
**Pull request by magodo**
fix #8577

### CODE REVIEW

**Comment by magodo** [comment]
The two failures in the [test result](https://ci-oss.hashicorp.engineering/buildConfiguration/TerraformOpenSource_TerraformProviders_AzureRMPublic_AZURERM_SERVICE_PUBLIC_AUTHORIZATION/148472) seems unrelated to my changes.

File: website/docs/r/role_definition.html.markdown [hunk 1]
```diff
@@ -47,9 +47,7 @@ The following arguments are supported:
 
 * `permissions` - (Required) A `permissions` block as defined below.
 
-* `assignable_scopes` - (Optional) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
-
-~> **NOTE:** The value for `scope` is automatically included in this list.
+* `assignable_scopes` - (Required) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
```
Comment (jackofallops) [inline website/docs/r/role_definition.html.markdown:51]:
```suggestion
* `assignable_scopes` - (Optional) One or more assignable scopes for this Role Definition, such as `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333`, `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup`, or `/subscriptions/0b1f6471-1bf0-4dda-aec3-111122223333/resourceGroups/myGroup/providers/Microsoft.Compute/virtualMachines/myVM`.
~> **NOTE:** The value for `scope` is automatically included in this list if no other values are supplied.
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @magodo 

Thanks for this PR, I've suggested a way to avoid a breaking change here that I believe should work correctly. It will require the tests to be updated also.

**Comment by magodo** [comment]
@jackofallops Thank you for your comments, I have updated the code and added another test for the case that no assinable scopes is specified. Please take another look!

**Review by katbyte — APPROVED** [review]
Thanks @magodo - this LGTM 👍

**Pull request merged**